File: Niveau 3/00 – Déblocage du niveau 3/1 - Emprunts de livres.py
nbLivres, nbJours = map(int, input().split(' '))
livres = [0] * nbLivres
for i in range(nbJours):
    nbClients = int(input())
    for j in range(nbClients):
        iLivre, duree = map(int, input().split(' '))
        if livres[iLivre] == 0:
            livres[iLivre] = duree
            print(1)
        else:
            print(0)
    for j in range(nbLivres):
        if(livres[j] != 0):
            livres[j] -= 1

File: Niveau 3/00 – Déblocage du niveau 3/1 - Emprunts de livres.cpp
#include <iostream>
using namespace std;
int main()
{
  int nbLivres, nbJours;
  cin >> nbLivres >> nbJours;
  int livres[nbLivres] = {0};
  for (int jour = 0; jour < nbJours; jour++)
  {
    int nbClientsToday;
    cin >> nbClientsToday;
    for (int client = 0; client < nbClientsToday; client++)
    {
      int idLivre, duree;
      cin >> idLivre >> duree;
      if (livres[idLivre] == 0)
      {
        cout << 1 << endl;
        livres[idLivre] = duree;
      }
      else
      {
        cout << 0 << endl;
      }
    }
    for (int idLivre = 0; idLivre < nbLivres; idLivre++)
    {
      if (livres[idLivre] != 0)
      {
        livres[idLivre]--;
      }
    }
  }
}
